File: RepoTransBench/target_projects/Python/Java/oxsecurity_MaskerLogger/src/test/java/com/oxsecurity/maskerlogger/publictests/TestPublicMaskedLogger.java
package com.oxsecurity.maskerlogger.publictests;

import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;

import java.util.regex.*;
import java.nio.file.Paths;
import java.io.File;
import java.util.logging.*;

import com.oxsecurity.maskerlogger.maskerformatter.MaskerFormatter;
import com.oxsecurity.maskerlogger.maskerformatter.MaskerFormatterJson;
import com.oxsecurity.maskerlogger.maskerformatter.AbstractMaskedLogger;

public class TestPublicMaskedLogger {

    private static final String TEST_CONFIG_PATH =
            Paths.get(new File(System.getProperty("user.dir")).getParent(),
                    "maskerlogger", "config", "gitleaks.toml").toString();

    static class DummyRecord extends LogRecord {
        public DummyRecord(String msg) {
            super(Level.SEVERE, msg);
            setLoggerName("dummy");
            setSourceClassName("public");
            setSourceMethodName("test");
        }
    }

    private MaskerFormatter formatter;
    private MaskerFormatterJson jsonFormatter;

    @BeforeEach
    public void setUp() {
        formatter = new MaskerFormatter("%(levelname)s: %(message)s", TEST_CONFIG_PATH, 33);
        jsonFormatter = new MaskerFormatterJson("%(message)s", TEST_CONFIG_PATH, 22);
    }

    @Test
    public void testMaskSecretLogicWithNewPattern() {
        AbstractMaskedLogger logger = new AbstractMaskedLogger(TEST_CONFIG_PATH);
        Pattern p = Pattern.compile("(x)(y+)");
        Matcher match = p.matcher("xyyyy");
        assertTrue(match.find());
        String msg = logger._maskSecret("xyyyy abc xyyyy", new Matcher[]{match});
        assertTrue(msg.chars().filter(c -> c == '*').count() > 0);
    }

    @Test
    public void testMaskSensitiveDataNoMatchNewmsg() {
        AbstractMaskedLogger logger = new AbstractMaskedLogger(TEST_CONFIG_PATH);
        DummyRecord record = new DummyRecord("totally safe entry");
        logger._maskSensitiveData(record);
        assertEquals("totally safe entry", record.getMessage());
    }

    @Test
    public void testMaskSensitiveDataWithMatchNewsecret() {
        AbstractMaskedLogger logger = new AbstractMaskedLogger(TEST_CONFIG_PATH);
        DummyRecord record = new DummyRecord("\"token\": \"abcd12345efgh\" and secret_key = zyxw");
        logger.setRedact(39);
        logger._maskSensitiveData(record);
        assertTrue(record.getMessage() instanceof String);
        assertFalse(record.getMessage().contains("abcd12345efgh"));
    }

    @Test
    public void testFormatterFullLogIntegrationPublic() {
        Logger logger = Logger.getLogger("logtest_public");
        logger.setLevel(Level.SEVERE);
        Handler handler = new StreamHandler(System.out, formatter);
        logger.addHandler(handler);
        try {
            logger.severe("\"another_key\": \"AIzaSoMEoth3rKEY344sdlGh289Ka3dLPd\"");
            logger.severe("\"AWS_SECRET_THISISFAKE\" and some more.");
            logger.severe("Datadog token is: 'zyxw9876zyxw9876zyxw9876zyxw9876'");
            logger.severe("\"pin\": \"5678\"");
        } finally {
            logger.removeHandler(handler);
        }
    }

    @Test
    public void testJsonFormatterLogrecordMaskingPublic() {
        DummyRecord rec = new DummyRecord("auth = \"FAKENEWSECRETXYZ\"");
        String value = jsonFormatter.format(rec);
        assertTrue(value.contains("auth"));
    }

    @Test
    public void testMaskerFormatterJsonSkipMaskPublic() {
        DummyRecord rec = new DummyRecord("publiclogtext");
        rec.setResourceBundleName("skip_mask"); // simulate apply_mask = false
        String result = jsonFormatter.format(rec);
        assertTrue(result.contains("publiclogtext"));
    }

    @Test
    public void testReprAndStrPublic() {
        String name = jsonFormatter.getClass().getSimpleName();
        assertTrue(name.contains("Json"));
    }
}

File: RepoTransBench/source_projects/Python/oxsecurity_MaskerLogger/public_tests/test_public_masked_logger.py
import unittest
import logging
import os
from maskerlogger.masker_formatter import MaskerFormatter, MaskerFormatterJson
from maskerlogger import masker_formatter

TEST_CONFIG_PATH = os.path.join(os.path.dirname(__file__), "..", "maskerlogger/config/gitleaks.toml")

class DummyRecord(logging.LogRecord):
    def __init__(self, msg):
        super().__init__(name="dummy", level=logging.ERROR, pathname=__file__, lineno=2, msg=msg, args=(), exc_info=None)

class TestMaskerLoggerFullPathsPublic(unittest.TestCase):
    def setUp(self):
        self.fmt = "%(levelname)s: %(message)s"
        self.formatter = MaskerFormatter(self.fmt, regex_config_path=TEST_CONFIG_PATH, redact=33)
        self.json_formatter = MaskerFormatterJson("%(message)s", regex_config_path=TEST_CONFIG_PATH, redact=22)

    def test_mask_secret_logic_with_new_pattern(self):
        logger = masker_formatter.AbstractMaskedLogger(TEST_CONFIG_PATH)
        # Simulate a different match with groups
        import re
        match = re.match(r"(x)(y+)", "xyyyy")
        msg = logger._mask_secret("xyyyy abc xyyyy", [match])
        self.assertTrue(msg.count("*") > 0)
    
    def test_mask_sensitive_data_no_match_newmsg(self):
        logger = masker_formatter.AbstractMaskedLogger(TEST_CONFIG_PATH)
        record = DummyRecord("totally safe entry")
        logger._mask_sensitive_data(record)
        self.assertEqual(record.msg, "totally safe entry")

    def test_mask_sensitive_data_with_match_newsecret(self):
        logger = masker_formatter.AbstractMaskedLogger(TEST_CONFIG_PATH)
        record = DummyRecord('"token": "abcd12345efgh" and secret_key = zyxw')
        logger.redact = 39
        logger._mask_sensitive_data(record)
        self.assertIsInstance(record.msg, str)
        self.assertFalse("abcd12345efgh" in record.msg)

    def test_formatter_full_log_integration_public(self):
        logger = logging.getLogger("logtest_public")
        logger.setLevel(logging.ERROR)
        handler = logging.StreamHandler()
        handler.setFormatter(self.formatter)
        logger.addHandler(handler)
        try:
            logger.error('"another_key": "AIzaSoMEoth3rKEY344sdlGh289Ka3dLPd"')
            logger.error('"AWS_SECRET_THISISFAKE" and some more.')
            logger.error("Datadog token is: 'zyxw9876zyxw9876zyxw9876zyxw9876'")
            logger.error('"pin": "5678"')
        finally:
            logger.removeHandler(handler)

    def test_json_formatter_logrecord_masking_public(self):
        rec = DummyRecord('auth = "FAKENEWSECRETXYZ"')
        value = self.json_formatter.format(rec)
        self.assertIn("auth", value)

    def test_maskerformatterjson_skip_mask_public(self):
        rec = DummyRecord("publiclogtext")
        setattr(rec, "apply_mask", False)
        result = self.json_formatter.format(rec)
        self.assertIn("publiclogtext", result)

    def test_repr_and_str_public(self):
        self.assertIn("Json", type(self.json_formatter).__name__)

if __name__ == "__main__":
    unittest.main()